src/vs/workbench/parts/debug/electron-browser/debugService.ts

```diff
@@ -439,6 +441,16 @@ export class DebugService implements debug.IDebugService {
 		});
 	}
 
+	private debouncedDisplayThreads(session: RawDebugSession) {
+		const timer = this.displayThreadsTimer.get(session.getId());
+		if (timer) {
+			clearTimeout(timer);
+		}
+		this.displayThreadsTimer.set(session.getId(), setTimeout(() => {
```

Comment:
Instead of a timeoue please use a `Scheduler`.
An example usage can be found here https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/parts/debug/electron-browser/callStackView.ts#L38